### ISSUE TITLE
Category path is the same as key producing duplicate URL issue

### ISSUE THREAD

**Issue by AaronKaa** [magento/magento2#14080]
### Preconditions
Magento 2.4
Ubuntu 18.04

### Steps to Reproduce

1. Create any category on second level (with parent_id =2).
Name: Test category 1, Url key: test-category-1.
2. Create sub category in "Test category 2" with `level = 2` by res api:
`/V1/categories`
```json
{
	"category": {
		"name": "Test category 2",
		"parent_id":"3",  /* Test category 1 */
		"is_active":"true",
		"include_in_menu": true,
		"level":"2",
		"custom_attributes": [{ "attribute_code": "url_key", "value": "test-category-2" }]
	}
}
```
At now we have the url_key: "test-category-1" and path: "/test-category-1/test-category-2".

3. Change the url key of the Test category 2 either in the backend or via the api to something like "test-category-2-changed"

### Expected result:
Category "Test category 2" has attribute values:
url_key: "test-category-2-changed"
url_path: "/test-category-1/test-category-2-changed"

### Actual result:
Category "Test category 2" has attribute values:
url_key: "test-category-2-changed"
url_path: "test-category-2-changed"

**Note**
When I try to change a category url key, it changes the path to be identical which results in an error being raised and short category urls which don't include its parent, it has now affected categories that I haven't explicitly changed/saved.
(table `catalog_category_entity_varchar`)

<img width="565" alt="screen shot 2018-03-13 at 12 35 48" src="https://user-images.githubusercontent.com/523347/37341961-214b6b5c-26bb-11e8-9423-9ac2a1f33ea0.png">

As you can see from the screen shot (with url_key, attribute_id 117 and url_path, attribute_id 118) the mens-footwear category has been affected but not the womens-footwear category.

**Comment by AaronKaa** [comment]
This looks like it could be related to issue #7298

**Comment by magento-engcom-team** [comment]
Hi @AaronKaa. Thank you for report.
We can not reproduce the issue.
Can you confirm that the issue is reproducing on clean magento installation?
Thanks

**Comment by AaronKaa** [comment]
I cant confirm that - the store itself has been updated from magento 2.1.5 and the data was imported via rest api (I should have mentioned that in the initial post), at this point were almost ready to go live with a new store.

**Comment by AaronKaa** [comment]
We've found that editing the category using the api doesn't have this effect so we're going to do it that way for now.

**Comment by magento-engcom-team** [comment]
@AaronKaa please more describe steps to reproduce the issue

**Comment by AaronKaa** [comment]
We imported all of the categories and products via the REST API on version 2.1.5 
Upgraded to 2.2.2
Tried to edit a category via both the web interface and the REST API and both produce the same results
Making a new category via the web interface and editing it doesn't have the same effect

**Comment by Lycanite** [comment]
I'm working with AaronKaa on this project. I think it's related to products added via the REST API.
For example, if you add a category with a parent category like this:
`{
    "category": {
        "name": "Test",
        "is_active": true,
        "include_in_menu": true,
        "level": LEVEL,
        "position": 0,
        "parent_id": PARENT_ID,
        "custom_attributes": [{
            "attribute_code": "url_key",
            "value": "test"
        }]
    }
}`

The url is at first correct, so say the parent category was called "Example" with the url key "example" we'd get: "/example/test".

But if we then try to change the url key of the test category either in the backend or via the api to something like "test-changed" we'd expect the new url path to be "/example/test-changed" but instead it becomes just "/test". So it looks like the url path is being set as the url key. This is happening with deeper categories as well.

Categories that were created manually in the backend and not with the api do not produce this result and work correctly however.

**Comment by magento-engcom-team** [comment]
Hi @AaronKaa and @Lycanite 
We were able to reproduce this issue by following steps:
1. Create any category on second level (with parent_id =2).
Name: Test category 1, Url key: test-category-1.
2. Crete sub category in "Test category 2" with `level = 2` by res api:
```json
{
	"category": {
		"name": "Test category 2",
		"parent_id":"3",  /* Test category 1 */
		"is_active":"true",
		"include_in_menu": true,
		"level":"2",
		"custom_attributes": [{ "attribute_code": "url_key", "value": "test-category-2" }]
	}
}
```
At now we have the url_key: "test-category-1" and path: "/test-category-1/test-category-2".

3. Change the url key of the Test category 2 either in the backend or via the api to something like ""test-category-2-changed"

### Expected result:
Category "Test category 2" has attribute values:
url_key: "test-category-2-changed"
path: "/test-category-1/test-category-2-changed"

### Actual result:
Category "Test category 2" has attribute values:
url_key: "test-category-2-changed"
path: "test-category-2-changed"

**Comment by magento-engcom-team** [comment]
@AaronKaa, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by AaronKaa** [comment]
Looking at this further Ive found that the issue is the category level - manually re-setting the level to its correct value in the database and regenerating the urls fixed the issue

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Echo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30130` were created

**Issue Available**: @engcom-Echo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @AaronKaa. Thank you for your report. 
The issue has been fixed in magento/magento2#27304 by @sergiy-v in 2.4-develop branch
Related commit(s):
 - [4da150c13e2540a886ddf1cfaf74ba9e2b167eea](https://github.com/magento/magento2/commit/4da150c13e2540a886ddf1cfaf74ba9e2b167eea)

The fix will be available with the upcoming 2.4.0 release.
